source/neuropod/tests/test_c_api.c

```diff
@@ -0,0 +1,162 @@
+/* Copyright (c) 2020 UATC, LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Inspired by the TensorFlow C API c_test.
+
+#include "neuropod/bindings/c/c_api.h"
+
+// This file exists just to verify that the header files above can build,
+// link, and run as "C" code.
+
+#ifdef __cplusplus
+#error "This file should be compiled as C code, not as C++."
+#endif
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+static void CheckFailed(const char *expression, const char *filename, int line_number)
+{
+    fprintf(stderr, "ERROR: CHECK failed: %s:%d: %s\n", filename, line_number, expression);
+    fflush(stderr);
+    abort();
+}
+
+// We use an extra level of macro indirection here to ensure that the
+// macro arguments get evaluated, so that in a call to CHECK(foo),
+// the call to STRINGIZE(condition) in the definition of the CHECK
+// macro results in the string "foo" rather than the string "condition".
+#define STRINGIZE(expression) STRINGIZE2(expression)
+#define STRINGIZE2(expression) #expression
+
+// Like assert(), but not dependent on NDEBUG.
+#define CHECK(condition) ((condition) ? (void) 0 : CheckFailed(STRINGIZE(condition), __FILE__, __LINE__))
+#define ASSERT_EQ(expected, actual) CHECK((expected) == (actual))
+#define ASSERT_NE(expected, actual) CHECK((expected) != (actual))
+#define ASSERT_STREQ(expected, actual) ASSERT_EQ(0, strcmp((expected), (actual)))
```

Comment:
If you didn't write this, can you include a comment with a link to where you got it from?

Reply:
This is pretty common implementation, but namely this one was copied from https://github.com/tensorflow/tensorflow/blob/master/tensorflow/lite/c/c_test.c . I add link to it.